apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,60 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
+  String variant
+
+  @InputFiles
+  Collection<File> getInputFiles() {
+    Set<File> inputFiles = Sets.newHashSet()
+    config.each { ext ->
+      ext.getFiles().asMap().entrySet().each { entry ->
+        entry.value.each { file ->
+          inputFiles.add(new File(entry.key, file))
+        }
+      }
+    }
+    return inputFiles
+  }
+
+  @Override
+  public void exec() {
```

Comment:
The following is equivalent running the following under `src/x/graphql` to:

`apollo-codgen **/**.graphql --schema schema.json --ouput VariantNameApi.json --target json`

A bunch of things to note here:

- This generates one IR for all the graphql files found under the directory. @martijnwalraven brought a good point about being able to reference fragments defined in other .graphql files.

- If we want to generate one per graphql, we'd have to basically run a shell script in a loop because gradle tasks custom can't instantiate other tasks. The solution would be a bit ugly since the class would no longer extend `NodeTask`.

I think a good solution to keep the process and maintain the package structure would be to have the relative path as a field in the IR. 

- As a consequence of the variants and sourceSet structure; this would actually do the following:
For every variant, grab all graphql files in its sourceSets. This means that for the debug variant, it would grab all graphql files under `src/main/graphql` and `src/debug/graphql`. 
Two things follow from this approach:

1. If duplicate operation names exist, apollo-codgen would fail. IMHO, it's safe to assume that won't happen in most of the cases since it defeats the purpose of having variants/flavors.

2. Where do we pick our schema file from? The flow right now would loop through the sourceSets directories and returns the first schema.json file it finds. IMO, it's also valid to assume that if you have multiple variants with different APIs, you're less likely to place your schema under the `main` sourceSet which is shared between variants.